### PR TITLE
update 2.0 public api in dataset&framework

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

Others
### PR changes

APIs
### Describe

update 2.0 public api in dataset&framework

背景：
对于用户调用api的层级，由于之前没有明确的设计，api开放层级比较随意。影响用户使用中的逻辑和正规感受。
目的：
1）重新整理api开放层级，提升用户使用感受
2）API保留自有特色，同时更好地符合行业规范，降低用户学习成本
3）提高框架api逻辑性，便于后续维护和优化
主要实现方法：
1）__all__内容清晰化，内容逐个列出，一行一个。不再使用 __all__ 的向上传递机制，仅在公开API层级添加 __all__，其余的 __all__ 列表统一置空
2）不再使用 #DEFINE_ALIAS
3）导入内容清晰化，原则上不再使用 import *， 按需引入需要的API， 每行一个
4）原则上不再使用 from . import module， from .module import submodule时会引入module，避免重复引用
5）添加新的Public API时，需向上寻找到最近的节点添加 import 并加入 __all__ 列表
6）Public API列表由多个公开API层级的 __all__ 列表共同组成，文档抽取以此为准，这些指定层级是：
- paddle
- paddle.amp
- paddle.nn
- paddle.nn.functional
- paddle.nn.initializer
- paddle.nn.utils
- paddle.static
- paddle.static.nn
- paddle.io
- paddle.jit
- paddle.metric
- paddle.distribution
- paddle.optimizer
- paddle.optimizer.lr
- paddle.regularizer
- paddle.text
- paddle.vision
- paddle.callbacks
- paddle.distributed
- paddle.distributed.fleet
- paddle.distributed.fleet.base.distributed_strategy
- paddle.distributed.fleet.data_generator
- paddle.distributed.fleet.utils
- paddle.distributed.parallel
- paddle.distributed.utils
- paddle.sysconfig
- paddle.utils
- paddle.utils.download
- paddle.utils.profiler
修改示例展示：
原来写法：
image
新的写法：
image

对开发者使用API的影响：
在指定层级推荐方式：（不受影响）
  from paddle.nn import CTCLoss
  from paddle.nn import Conv2D
非指定层级支持但不推荐：（不受影响）
  from paddle.nn.layer.loss import CTCLoss
  from paddle.nn.layer.conv import Conv2D
对于import *使用方式不推荐。（非公开层级受影响）
  不再支持：（非公开层级）
  from paddle.nn.layer import *
  from paddle.nn.layer.conv import *
  from paddle.nn.initializer.norm import *
  from paddle.tensor.math import *
  from paddle.optimizer.adam import *
  可以支持：（公开层级）
  from paddle import *
  from paddle.nn import *
  from paddle.utils import *
  from paddle.optimizer import *

本pr相关修改对api的调用无明显影响

### CODE REVIEW

File: python/paddle/dataset/__init__.py [hunk 1]
```diff
@@ -15,18 +15,18 @@
...
+import paddle.dataset.movielens  # noqa: F401
+import paddle.dataset.conll05  # noqa: F401
+import paddle.dataset.uci_housing  # noqa: F401
+import paddle.dataset.wmt14  # noqa: F401
+import paddle.dataset.wmt16  # noqa: F401
+import paddle.dataset.flowers  # noqa: F401
+import paddle.dataset.voc2012  # noqa: F401
+import paddle.dataset.image  # noqa: F401
 
 # set __all__ as empty for not showing APIs under paddle.dataset
```
Comment (willthefrog) [inline python/paddle/dataset/__init__.py:31]: why?

Comment (zhiboniu) [reply python/paddle/dataset/__init__.py:31]: 原来就已经在这了，我也不清楚设计者的目的，要移除吗？

Comment (willthefrog) [reply python/paddle/dataset/__init__.py:31]: 搞搞清楚啊。。

Comment (zhiboniu) [reply python/paddle/dataset/__init__.py:31]: 这些是原来静态图迭代式的接口，现在统一用DataLoader里的Dataset式的接口了，所以deprecate了

Comment (iclementine) [reply python/paddle/dataset/__init__.py:31]: 这里有点奇怪，原来就不是用 from ... import ... 的用法而是 import 全名。虽然因为是在 dataset/__init__.py 里这么写所以一样。

File: python/paddle/framework/__init__.py [hunk 2]
```diff
@@ -12,31 +12,24 @@
...
-from .random import seed
-from .framework import get_default_dtype
-from .framework import set_default_dtype
+from ..fluid.param_attr import ParamAttr  # noqa: F401
+from ..fluid.layers.tensor import create_parameter  # noqa: F401
+from ..fluid.core import CPUPlace  # noqa: F401
+from ..fluid.core import CUDAPlace  # noqa: F401
+from ..fluid.core import CUDAPinnedPlace  # noqa: F401
+from ..fluid.core import NPUPlace  # noqa: F401
+from ..fluid.core import VarBase  # noqa: F401
```
Comment (iclementine) [inline python/paddle/framework/__init__.py:29]: 这里需要 as Tensor 吗？

Comment (iclementine) [reply python/paddle/framework/__init__.py:29]: 以及还有 paddle/__init__.py 下面有一个 
from .framework import VarBase as Tensor

Comment (zhiboniu) [reply python/paddle/framework/__init__.py:29]: 是不是paddle.__init__下的公开api用Tensor就好了？这里是内部代码，是否有必要也都修改了，改太多怕容易改出错

Comment (iclementine) [reply python/paddle/framework/__init__.py:29]: okay

**Review by iclementine — DISMISSED** [review]
LGTM for current state, but conflicts need to be resolved.

**Review by jzhang533 — APPROVED** [review]
LGTM

File: python/paddle/__init__.py [hunk 3]
```diff
@@ -264,6 +264,7 @@
 
 from .framework import set_default_dtype  #DEFINE_ALIAS
 from .framework import get_default_dtype  #DEFINE_ALIAS
+from .framework import set_grad_enabled  #DEFINE_ALIAS
```
Comment (XiaoguangHu01) [inline python/paddle/__init__.py:267]: 这个paddle/__init__.py文件怎么没有定义__all__列表呢？

Comment (zhiboniu) [reply python/paddle/__init__.py:267]: 这个文件有一个单独的pr，31985这里只是在paddle.init.py里做了和framework相关的更新。
paddle.init.py的pr:
https://github.com/PaddlePaddle/Paddle/pull/32034

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by phlrain — APPROVED** [review]
